src/components/PopupWithForm.js

```diff
@@ -1,34 +1,42 @@
 import Popup from "./Popup.js";
-import FormValidator from "./FormValidator.js";
 
 export default class PopupWithForm extends Popup {
-  constructor({ popupSelector, handleFormSubmit }) {
-    super({ popupSelector });
-    this._popupForm = this._popupElement.querySelector(".modal__form");
-    this._popupInputs = this._popupForm.querySelectorAll(".modal__input");
+  constructor(popupSelector, handleFormSubmit) {
+    super(popupSelector);
+    this._submitButton = this._popupElement.querySelector(".modal__button");
     this._handleFormSubmit = handleFormSubmit;
+    this._popupForm = this._popupElement.querySelector(".modal__form");
+    this._inputList = this._popupForm.querySelectorAll(".modal__input");
+    this._submitButtonText = this._submitButton.textContent;
   }
 
   _getInputValues() {
     const formValues = {};
-    this._popupInputs.forEach((input) => {
+    this._inputList.forEach((input) => {
       formValues[input.name] = input.value;
     });
-
     return formValues;
   }
 
-  _setInputValues(data) {
-    this._inputList.forEach((input) => {
-      input.value = data[input.name];
-    });
+  reset() {
+    this._popupForm.reset();
   }
+
```

Comment:
**Could be improved**

You can make ` setInputValues` in class `PopupWithForm`, which can insert data into inputs:

```js

  setInputValues(data) {
    this._inputList.forEach((input) => {
      // Here you insert the `value` by the `name` of the input
      input.value = data[input.name];
    });
  }



```

So, this way you’ll not have to search the inputs of the profile in `index.js`